src/richtext/utils/hastUtilToRichText.ts

```diff
@@ -0,0 +1,373 @@
+ * @remarks
+ * The `span` rich text node type is not available as it is not relevant in the
+ * context of going from HTML to Prismic rich text.
+ */
+type RichTextHTMLMapSerializerShorthand =
+	| Exclude<RichTextNodeTypes, "label" | "span">
+	| { label: string };
+
+/**
+ * The payload provided to a {@link RichTextHTMLMapSerializerFunction}.
+ */
+type RichTextHTMLMapSerializerFunctionPayload = {
+	/**
+	 * The hast {@link Element} node to serialize.
+	 */
+	node: Element;
+
+	/**
+	 * Additional context information to help with the serialization.
+	 */
+	context: {
+		/**
+		 * The list type of the last list node encountered if any.
+		 */
+		listType: "group-list-item" | "group-o-list-item" | null;
+	};
+};
+
+/**
+ * Serializes a hast {@link Element} node to a
+ * {@link RichTextHTMLMapSerializerShorthand} or a rich text node.
+ *
+ * @remarks
+ * Serializing to a rich text node directly is not recommended and is only
+ * available as an escape hatch. Prefer returning a
+ * {@link RichTextHTMLMapSerializerShorthand} instead.
+ */
+type RichTextHTMLMapSerializerFunction = (
+	payload: RichTextHTMLMapSerializerFunctionPayload,
+) =>
+	| RichTextHTMLMapSerializerShorthand
+	| RTNode
+	| RTInlineNode
+	| RTPartialInlineNode
+	| null
+	| undefined;
+
+/**
+ * Serializes an hast {@link Element} node matching the given HTML tag name or
+ * CSS selector to a Prismic rich text node.
+ */
+export type RichTextHTMLMapSerializer = Record<
+	string,
+	RichTextHTMLMapSerializerShorthand | RichTextHTMLMapSerializerFunction
+>;
+
+const DEFAULT_SERIALIZER: RichTextHTMLMapSerializer = {
+	h1: "heading1",
+	h2: "heading2",
+	h3: "heading3",
+	h4: "heading4",
+	h5: "heading5",
+	h6: "heading6",
+	p: "paragraph",
+	pre: "preformatted",
+	strong: "strong",
+	em: "em",
+	li: ({ context }) =>
+		context.listType === "group-o-list-item" ? "o-list-item" : "list-item",
+	ul: "group-list-item",
+	ol: "group-o-list-item",
+	img: "image",
+	iframe: "embed",
+	a: "hyperlink",
+};
+
+const VFILE_RULE = "failed-to-serialize-node";
+const VFILE_SOURCE = "prismic";
+
+/**
+ * Configuration that determines the output of `hastUtilToRichText`.
+ */
+export type HastUtilToRichTextConfig = {
+	/**
+	 * An optional HTML to rich text serializer. Will be merged with the default
+	 * HTML to rich text serializer.
+	 */
+	serializer?: RichTextHTMLMapSerializer;
+
+	/**
+	 * Whether or not the text processed should be marked as right-to-left.
+	 *
+	 * @defaultValue `false`
+	 */
+	direction?: "ltr" | "rtl";
+
+	/**
+	 * The default node type to wrap the input with in case it only represents the
+	 * inner content of an element.
+	 *
+	 * @remarks
+	 * This is a niche option that allows you to serialize correctly an input such
+	 * as: `"lorem <strong>ipsum</strong> dolor sit amet"`, wrapping it inside a
+	 * node of the desired type.
+	 *
+	 * @defaultValue `"paragraph"`
+	 */
+	defaultWrapperNodeType?: RTTextNode["type"];
+};
+
+export const hastUtilToRichText = (
```

Comment:
Happy to rename it, this was an attempt at being somewhat consistent with what hast is doing for `hast-util-to-html`: https://github.com/syntax-tree/hast-util-to-html

Reply:
Renamed the export to `toRichText` to align with `hast-util-to-html` while keeping the same file name, let me know what you think!

Reply:
`hastToRichText` is clearer in my opinion, along with a `hastToRichText.ts` filename. I think `toHtml` works for `hast-util-to-html` because the function name has the `hast-util-to-html` package name as context. In our case, we don't have the same kind of limited context.